core/base/src/TROOT.cxx

```diff
@@ -226,10 +226,37 @@ static Int_t ITIMQQ(const char *time)
    return 100*hh + mm;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Write and Close all open writable TFiles, useful to be called when SIGTERM is caught.
+
+void TROOT::WriteCloseAllFiles()
+{
+   if (gROOT) {
+      R__LOCKGUARD(gROOTMutex);
+
+      if (gROOT->GetListOfFiles()) {
+         TIter next(gROOT->GetListOfFiles());
+         while (TObject *obj = next()) {
+            if (obj && obj->InheritsFrom(TClass::GetClass("TFile", kFALSE, kTRUE))) {
+               TMethodCall callIsWritable(obj->IsA(), "IsWritable", "");
```

Comment:
Why is `TMethodCall` needed and a cast is not enough?

Reply:
We would need to cast to TFile and thus `#include "TFile.h"`, which would introduce a dependency of libCore from libRIO, which I guess is not ideal? The only other solution would be to create a baseclass in Core from which T(Directory)File derives that implements IsWritable and is overridden in the derived.

Reply:
It is already in `TDirectory` ...